lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2622,6 +2628,33 @@ ParseResult FIRStmtParser::parseRefReleaseInitial() {
   return success();
 }
 
+/// connect ::= 'connect' expr expr
+ParseResult FIRStmtParser::parseConnect() {
+  auto startTok = consumeToken(FIRToken::kw_connect);
+
+  Value lhs, rhs;
+  if (parseExp(lhs, "expected connect expression") ||
+      parseExp(rhs, "expected connect expression") || parseOptionalInfo())
+    return failure();
+
+  locationProcessor.setLoc(startTok.getLoc());
+  emitConnect(builder, lhs, rhs);
```

Comment:
Can this check the same type-equivalence done on the other path for friendly errors?

Reply:
Good idea. I'll fix this.

Reply:
Fixed in: https://github.com/llvm/circt/commit/0bb6ec6212f61330aba457e29e8cc91c2f009687